app/src/androidTest/java/org/mozilla/focus/autobot/Autobot.kt

```diff
@@ -77,6 +79,30 @@ class ScreenshotRobot : MenuRobot() {
     }
 }
 
+class FindInPageRobot : MenuRobot() {
+
+    fun findKeywordInPage(keyword: String) {
+
+        onView(withId(R.id.find_in_page_query_text)).check(matches(isDisplayed()))
+        onView(withId(R.id.find_in_page_query_text)).perform(replaceText(keyword))
+    }
+
+    fun navigateKeywordSearchInPage(forward: Boolean = false) {
+//        // default action is to view the next search result
+//        onView(withId(R.id.find_in_page_result_text)).check(matches(isDisplayed()))
+//        onView(withId(R.id.find_in_page_prev_btn)).check(matches(isDisplayed()))
+//        onView(withId(R.id.find_in_page_next_btn)).check(matches(isDisplayed()))
+//        onView(withId(R.id.find_in_page_close_btn)).check(matches(isDisplayed()))
+        if (forward) onView(withId(R.id.find_in_page_prev_btn)).perform(click())
+        else onView(withId(R.id.find_in_page_next_btn)).perform(click())
+    }
+
+    fun closeButtonForFindInPage() {
+        onView(withId(R.id.find_in_page_close_btn)).check(matches(isDisplayed()))
```

Comment:
I didn't get it. What's the purpose to tap on url bar here?

Reply:
that is an example. what i mean is line 101 and line 102 can merge to 
`onView(Matchers.allOf(withId(R.id.find_in_page_close_btn), isDisplayed())).perform(click())`
which will check whether it is displayed or not. If it is displayed it will perform click().